movies_metadata.csv Dataset downloaded from Kaggle.
https://www.kaggle.com/rounakbanik/the-movies-dataset/version/7

In [26]:
# Import plotly offline
import plotly
plotly.offline.init_notebook_mode(connected=True)
import plotly.graph_objs as go
import numpy as np
import pandas as pd

# Import pandas
import pandas as pd

In [240]:
movies = pd.read_csv('movies_metadata.csv', low_memory=False)

In [66]:
print (movies.head())

   adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497         

In [241]:
# Much more data than we need.
# Select just original title, years, budget, revenue, and length
edit_movies_df = movies.drop(['adult',
                              'belongs_to_collection',
                              'genres','homepage',
                              'id','imdb_id',
                              'original_language',
                              'original_title',
                              'overview','popularity',
                              'poster_path','production_companies',
                              'production_countries','spoken_languages',
                              'status','tagline','video','vote_average',
                              'vote_count'], axis=1)

In [252]:
edit_movies_df.head()

budget release_date      revenue  runtime                        title
0  30000000   1995-10-30  373554033.0     81.0                    Toy Story
1  65000000   1995-12-15  262797249.0    104.0                      Jumanji
2         0   1995-12-22          0.0    101.0             Grumpier Old Men
3  16000000   1995-12-22   81452156.0    127.0            Waiting to Exhale
4         0   1995-02-10   76578911.0    106.0  Father of the Bride Part II

In [255]:
# Remove all non numeric containing rows.
edit_movies_df = edit_movies_df[pd.to_numeric(edit_movies_df['budget'], errors='coerce').notnull()]
edit_movies_df = edit_movies_df[pd.to_numeric(edit_movies_df['revenue'], errors='coerce').notnull()]
edit_movies_df = edit_movies_df[pd.to_numeric(edit_movies_df['runtime'], errors='coerce').notnull()]

In [256]:
# Convert all numeric columns to floats, in case they are strings.
edit_movies_df['budget'] = edit_movies_df['budget'].astype('float64')
edit_movies_df['revenue'] = edit_movies_df['revenue'].astype('float64')
edit_movies_df['runtime'] = edit_movies_df['runtime'].astype('float64')
edit_movies_df.head()

budget release_date      revenue  runtime                        title
0  30000000.0   1995-10-30  373554033.0     81.0                    Toy Story
1  65000000.0   1995-12-15  262797249.0    104.0                      Jumanji
2         0.0   1995-12-22          0.0    101.0             Grumpier Old Men
3  16000000.0   1995-12-22   81452156.0    127.0            Waiting to Exhale
4         0.0   1995-02-10   76578911.0    106.0  Father of the Bride Part II

In [257]:
# Drop all NaN or 0 containing rows
edit_movies_df = edit_movies_df.replace(0, np.nan).replace(0.0, np.nan).dropna()

In [258]:
edit_movies_df['release_date'] = edit_movies_df['release_date'].apply(lambda date: pd.to_datetime(date))

In [259]:
# How many movies left in our list?
print (len(edit_movies_df))
edit_movies_df.head()

5369


budget release_date      revenue  runtime              title
0  30000000.0   1995-10-30  373554033.0     81.0          Toy Story
1  65000000.0   1995-12-15  262797249.0    104.0            Jumanji
3  16000000.0   1995-12-22   81452156.0    127.0  Waiting to Exhale
5  60000000.0   1995-12-15  187436818.0    170.0               Heat
8  35000000.0   1995-12-22   64350171.0    106.0       Sudden Death

In [260]:
# Group movies by year, this is how we'll be sorting them later
date_sort = edit_movies_df.sort_values('release_date')
print (date_sort.head())

         budget release_date     revenue  runtime  \
6913   100000.0   1915-02-08  11000000.0    193.0   
8293    17311.0   1915-12-13    137365.0     59.0   
13003  200000.0   1916-12-24   8000000.0    105.0   
35914  250000.0   1918-08-01   8000000.0     93.0   
3192   250000.0   1921-01-21   2500000.0     68.0   

                              title  
6913          The Birth of a Nation  
8293                      The Cheat  
13003  20,000 Leagues Under the Sea  
35914                        Mickey  
3192                        The Kid  


In [261]:
# Assign dates to list
date_list = date_sort['release_date'].tolist()

# Assign revenue to list
revenue_list = date_sort['revenue'].tolist()

# Assign budget to list
budget_list = date_sort['budget'].tolist()

# Assign runtime to list
runtime_list = date_sort['runtime'].tolist()

In [264]:
# Graph using plotly, after generating series of x and y axis
trace1 = go.Bar(x = date_list, y = budget_list, name = 'Budget')

data = [trace1]

layout = dict(title = 'Movie Budget Over Time',
              xaxis = dict(title = 'Date'),
              yaxis = dict(title = 'Dollars'),
              )

fig = dict(data=data, layout=layout)
plotly.offline.iplot(fig, filename='Movie Budget Over Time')


In [265]:
# Graph using plotly, after generating series of x and y axis
trace2 = go.Bar(x = date_list, y = revenue_list, name = 'Revenue')

data = [trace2]

layout = dict(title = 'Movie Revenue Over Time',
              xaxis = dict(title = 'Date'),
              yaxis = dict(title = 'Dollars'),
              )

fig = dict(data=data, layout=layout)
plotly.offline.iplot(fig, filename='Movie Revenue Over Time')


In [263]:
# Graph using plotly, after generating series of x and y axis
trace3 = go.Bar(x = date_list, y = runtime_list, name = 'Runtime')

data = [trace3]

layout = dict(title = 'Movie Runtime Over Time',
              xaxis = dict(title = 'Date'),
              yaxis = dict(title = 'Minutes'),
              )

fig = dict(data=data, layout=layout)
plotly.offline.iplot(fig, filename='Movie Runtime Over Time')


In [266]:
# This data is interesting, but mean for each year would be better
date_sort['YEAR'] = date_sort['release_date'].apply(lambda x: x.strftime('%Y'))

In [267]:
date_sort.head()

budget release_date     revenue  runtime  \
6913   100000.0   1915-02-08  11000000.0    193.0   
8293    17311.0   1915-12-13    137365.0     59.0   
13003  200000.0   1916-12-24   8000000.0    105.0   
35914  250000.0   1918-08-01   8000000.0     93.0   
3192   250000.0   1921-01-21   2500000.0     68.0   

                              title  YEAR  
6913          The Birth of a Nation  1915  
8293                      The Cheat  1915  
13003  20,000 Leagues Under the Sea  1916  
35914                        Mickey  1918  
3192                        The Kid  1921

In [308]:
# EXAMINING SPECIFIC YEARS

# Let's test what movies were released during specific years,
# To see what causes the spikes we see on the average graph.
specific_year = date_sort.loc[date_sort['YEAR'] == '2017']
specific_year


budget release_date       revenue  runtime  \
41928   30000000.0   2017-01-12  3.254067e+07     95.0   
41973    7400000.0   2017-01-12  2.666720e+07     96.0   
41972   85000000.0   2017-01-13  3.461477e+08    107.0   
44791    8520000.0   2017-01-18  5.610000e+07    125.0   
42411    4000000.0   2017-01-19  1.440000e+06     90.0   
41814   22000000.0   2017-01-19  1.946473e+08    100.0   
43058    5000000.0   2017-01-20  3.305592e+06     87.0   
41488   30000000.0   2017-01-26  1.479338e+07    121.0   
42175   18000000.0   2017-01-27  4.141271e+07    117.0   
42018   65000000.0   2017-01-27  2.542122e+08    130.0   
42096   25000000.0   2017-02-01  8.308089e+07    102.0   
41976   80000000.0   2017-02-08  3.119504e+08    104.0   
42168   40000000.0   2017-02-08  1.715399e+08    122.0   
42355   55000000.0   2017-02-08  3.788275e+08    118.0   
41929   40000000.0   2017-02-15  2.655956e+07    146.0   
43499     707503.0   2017-02-16  4.781454e+06     90.0   
43101   20000000.0   2017-02-23  5.572410e+05    103.0   
42169    5000000.0   2017-02-24  2.524342e+08    104.0   
42170   97000000.0   2017-02-28  6.168018e+08    137.0   
42185    5000000.0   2017-03-02  1.002557e+07     98.0   
42846   20000000.0   2017-03-03  5.733087e+07    132.0   
42171  185000000.0   2017-03-08  5.666528e+08    118.0   
42805   40000000.0   2017-03-10  2.409780e+05    111.0   
42222  160000000.0   2017-03-16  1.262886e+09    129.0   
42536   10000000.0   2017-03-17  4.436840e+05    129.0   
42853   58000000.0   2017-03-22  1.005418e+08    103.0   
43057   25000000.0   2017-03-23  2.680015e+07    100.0   
42901  100000000.0   2017-03-23  1.423372e+08    124.0   
43156   60000000.0   2017-03-23  1.971835e+08     89.0   
42235  125000000.0   2017-03-23  4.988149e+08     97.0   
...            ...          ...           ...      ...   
43353    2800000.0   2017-04-21  3.600000e+06     80.0   
42214  175000000.0   2017-04-27  1.461751e+08    126.0   
43575   18000000.0   2017-04-27  2.300000e+08    161.0   
33361   18000000.0   2017-04-27  2.049784e+07    110.0   
43371     916000.0   2017-04-28  2.403885e+06     99.0   
43233   42000000.0   2017-05-03  6.084571e+07     90.0   
42902   97000000.0   2017-05-09  2.323802e+08    122.0   
43231   69000000.0   2017-05-12  1.778568e+08    116.0   
43221   22000000.0   2017-05-19  4.012014e+07     91.0   
26560  230000000.0   2017-05-23  7.941920e+08    129.0   
33356  149000000.0   2017-05-30  8.205804e+08    141.0   
43286   38000000.0   2017-06-01  1.108244e+08     89.0   
43238  125000000.0   2017-06-06  4.077780e+08    110.0   
39694    5500000.0   2017-06-15  4.423502e+07     89.0   
43294  175000000.0   2017-06-15  3.501701e+08    109.0   
44009   80000000.0   2017-06-15  1.020063e+09     96.0   
43691   20000000.0   2017-06-15  4.505677e+07    101.0   
43318   10000000.0   2017-06-16  4.219536e+06    105.0   
44842  260000000.0   2017-06-21  6.049421e+08    149.0   
43642   10500000.0   2017-06-23  2.544294e+07     93.0   
43644   34000000.0   2017-06-28  2.245113e+08    113.0   
42308     100000.0   2017-07-07  1.558426e+07     92.0   
44274  152000000.0   2017-07-11  3.699080e+08    140.0   
44678  100000000.0   2017-07-19  5.198769e+08    107.0   
44337  197471676.0   2017-07-20  9.002429e+07    137.0   
44632   30000000.0   2017-07-26  9.000794e+07    115.0   
45139   50000000.0   2017-07-28  6.691394e+07     86.0   
45014   60000000.0   2017-08-03  7.100000e+07     95.0   
45167   11000000.0   2017-08-03  1.847702e+08    111.0   
44282   21000000.0   2017-08-04  2.452716e+07     95.0   

                                                  title  YEAR  
41928                                         Sleepless  2017  
41973                                   The Bye Bye Man  2017  
41972                        xXx: Return of Xander Cage  2017  
44791                           Confidential Assignment  2017  
42411                                        Black Snow  2017  
41814                          

In [270]:
# Group by years, find means
years_mean_df = date_sort.groupby('YEAR', as_index=True).mean()
#years_mean_df = years_mean_df.mean().sort_values('YEAR')

In [271]:
years_mean_df.head()

budget    revenue  runtime
YEAR                               
1915    58655.5  5568682.5    126.0
1916   200000.0  8000000.0    105.0
1918   250000.0  8000000.0     93.0
1921   250000.0  2500000.0     68.0
1922  1100000.0   400200.0    117.0

In [272]:
# Generate YEAR_list, from original date_sort
YEAR_list = date_sort['YEAR'].drop_duplicates().tolist()
type(YEAR_list)
print (YEAR_list[0:10])

['1915', '1916', '1918', '1921', '1922', '1924', '1925', '1926', '1927', '1928']


In [273]:
# Create new variable lists for x and y axis
# Generate YEAR_list, from original date_sort
YEAR_list = date_sort['YEAR'].drop_duplicates().tolist()

# Assign revenue to list
revenue_list_2 = years_mean_df['revenue'].tolist()

# Assign budget to list
budget_list_2 = years_mean_df['budget'].tolist()

# Assign runtime to list
runtime_list_2 = years_mean_df['runtime'].tolist()

In [274]:
print (YEAR_list[0:10])
print (revenue_list_2[0:10])
print (budget_list_2[0:10])
print (runtime_list_2[0:10])

['1915', '1916', '1918', '1921', '1922', '1924', '1925', '1926', '1927', '1928']
[5568682.5, 8000000.0, 8000000.0, 2500000.0, 400200.0, 1213880.0, 6840800.2000000002, 812439.0, 839089.5, 679387.75]
[58655.5, 200000.0, 250000.0, 250000.0, 1100000.0, 1135654.0, 1034318.4, 418227.5, 46551551.5, 156084.25]
[126.0, 105.0, 93.0, 68.0, 117.0, 149.0, 122.40000000000001, 98.0, 116.5, 74.0]


In [275]:
# Graph using plotly, after generating series of x and y axis
trace1 = go.Scatter(x = YEAR_list, y = revenue_list_2, name = 'Revenue')
trace2 = go.Scatter(x = YEAR_list, y = budget_list_2, name = 'Budget')
data = [trace1, trace2]

layout = dict(title = 'Average Movie Revenue and Budget By Year',
              xaxis = dict(title = 'Year'),
              yaxis = dict(title = 'Dollars'),
              )

fig = dict(data=data, layout=layout)
plotly.offline.iplot(fig, filename='Movie Budget Over Time')


Revenue represents total lifetime gross, not just box office sales while in theatres.  This may explain the spikes with early years.  These years contain big box office movies that have also seen re-release and sucess on home video or DVD, and once we removed rows with zeroes some years have fewer movies.  1937 is only represented by "Snow White and the Seven Dwarfs", which has enjoyed massive success in re-release and home video.  1939 was when "Gone with the Wind" and "The Wizard of Oz" were released, 1942 was "Bambi" and "Casablanca".  A later spike, 1977, was when "Star Wars" came out, another film that's had re-release in theatres and massive success on VHS and DVD.

1927's revenue mean seems to be skewed by the reported budget of "Metropolis" in this dataset, ~92 million US dollars.  The budget I found reported online was 5.3 million Reichsmarks, which converts to roughly 18 million US dollars today.  I'm not sure where the source of this dataset got their budget.

Looking at the raw graphs, another outlier is in 1962.  This corresponds with "The Manchurian Candidate".  Again, this dataset reports a higher budget than what is found online through other sources.  This dataset reports 80 million US dollars, when the budget is reported as 2.2 million online.  Even when converted to adjust for inflation, this only equals 18 million US dollars.

Sources:

Metropolis Budget: https://en.wikipedia.org/wiki/Metropolis_(1927_film)

Historical Exchange Rate: http://www.history.ucsb.edu/faculty/marcuse/projects/currency.htm

Conversion to 2018 dollars: http://www.in2013dollars.com/1927-dollars-in-2018?amount=1258907

Manchurian Candidate Budget: https://www.space.ca/the-manchurian-candidate-tiff-restored/


In [276]:
# Average runtime each year
# Graph using plotly, after generating series of x and y axis
trace3 = go.Scatter(x = YEAR_list, y = runtime_list_2, name = 'Runtime')

data = [trace3]

layout = dict(title = 'Average Movie Runtime By Year',
              xaxis = dict(title = 'Year'),
              yaxis = dict(title = 'Minutes'),
              )

fig = dict(data=data, layout=layout)
plotly.offline.iplot(fig, filename='Movie Runtime Over Time')

Why are the runtime means so much more variable?  After removing rows containing 0 or NaNs, many early years are only represented by one movie.  This causes radical variability in mean.  Once we reach the 1970s there are films with more complete information.  Then we see a fairly stable mean of ~110 minutes from the 1980s up through the present.